### PR TITLE
ceph ansible teardown and clients-role

### PR DESCRIPTION
**Pull request by vasukulkarni**
adds setup clients functions for old ansible builds and teardown for non vps nodes.

### CODE REVIEW

**Comment by zmc** [comment]
test this please

File: teuthology/task/ceph_ansible.py [hunk 1]
```diff
@@ -210,6 +215,56 @@ def teardown(self):
         os.remove(self.inventory)
         os.remove(self.playbook_file)
         os.remove(self.extra_vars_file)
+        machine_type = self.ctx.config('machine_type')
```
Comment (zmc) [inline teuthology/task/ceph_ansible.py]: This is not going to be reliable. You'll probably want to use ``teuthology.lock.query.is_vm`` or ``Remote.is_vm``

File: teuthology/task/ceph_ansible.py [hunk 2]
```diff
@@ -210,6 +215,56 @@ def teardown(self):
...
+            time.sleep(4)
+            self.ctx.cluster.run(args=['sudo', 'stop', 'ceph-all'],
+                                 check_status=False)
+            installer_node = self.installer_node
+            installer_node.run(args=['rm', '-rf', 'ceph-ansible'])
+            remove_osd_mounts(self.ctx)
+            remove_ceph_packages(self.ctx)
+            if self.config.get('rhbuild'):
+                if installer_node.os.package_type == 'rpm':
+                    installer_node.run(args=[
```
Comment (zmc) [inline teuthology/task/ceph_ansible.py]: Why not use the packaging module for this?

File: teuthology/task/ceph_ansible.py [hunk 3]
```diff
@@ -210,6 +215,56 @@ def teardown(self):
...
+                    'rm',
+                    '-rf',
+                    run.Raw('/etc/systemd/system/ceph*')],
+                check_status=False)
+            self.ctx.cluster.run(
+                args=[
+                    'sudo',
+                    'rm',
+                    '-rf',
+                    run.Raw('/etc/systemd/system/multi-user.target.wants/ceph*')],
```
Comment (zmc) [inline teuthology/task/ceph_ansible.py]: This could be simply added to the previous invocation

File: teuthology/task/ceph_ansible.py [hunk 4]
```diff
@@ -358,6 +418,45 @@ def run_playbook(self):
         # need to fix the permission on keyring to be readable by them
         self.fix_keyring_permission()
 
+    def setup_client_node(self):
```
Comment (zmc) [inline teuthology/task/ceph_ansible.py]: ceph-ansible doesn't do this?

File: teuthology/task/ceph_ansible.py [hunk 5]
```diff
@@ -59,6 +61,9 @@ class CephAnsible(Task):
         repo: {git_base}ceph-ansible.git
         branch: mybranch # defaults to master
         ansible-version: 2.2 # defaults to 2.1
+        # for old ansible version where clients roles
```
Comment (zmc) [inline teuthology/task/ceph_ansible.py]: 'ceph-ansible' and which versions?

File: teuthology/task/ceph_ansible.py [hunk 6]
```diff
@@ -275,6 +330,11 @@ def run_rh_playbook(self):
         if re.search(r'all hosts have already failed', out.getvalue()):
             log.error("Failed during ceph-ansible execution")
             raise CephAnsibleError("Failed during ceph-ansible execution")
+        # old ansible doesn't have clients role, setup clients for those
+        # cases
+        if self.config.get('setup-client'):
```
Comment (zmc) [inline teuthology/task/ceph_ansible.py]: 'setup-client' or 'setup-clients' ?

**Comment by kshtsk** [comment]
@vasukulkarni is this PR still actual?

**Pull request closed without merging**